### ISSUE TITLE
Why PCL?

### ISSUE THREAD

**Issue by Mike-DE-RE** [RWAP/PrinterToPDF#7]
> It is hoped that support for PCL (and potentially ESC/POS) could also be added

Are you aware of ghostpcl?

https://www.ghostscript.com/GhostPCL.html

This is a powerful tool to convert pcl files. I've use it a lot and find it very helpful.

**Comment by RWAP** [comment]
Yes - the problem is that GhostPCL is too expensive to use in our commercial product (it is more than we will be charging for the whole product) - so whilst users could install GhostPCL themselves, we can only dump the PCL file to disk for them to later convert which is not as nice 

I cannot find the cost information anywhere - but see [this](http://www.randomhacks.co.uk/how-much-does-it-cost-to-distribute-ghostscript-commercially/) for the costs in 2014

**Comment by Mike-DE-RE** [comment]
"If I use a piece of software that has been obtained under the GNU GPL, am I allowed to modify the original code into a new program, then distribute and sell that new program commercially? (#GPLCommercially)

You are allowed to sell copies of the modified program commercially, but only under the terms of the GNU GPL. Thus, for instance, you must make the source code available to the users of the program as described in the GPL, and they must be allowed to redistribute and modify it as escribed in the GPL.

These requirements are the condition for including the GPL-covered code you received in a program of your own."

http://www.gnu.org/licenses/gpl-faq#GPLCommercially

I have no idea what you intend to sell. Does this meet your requirements?

**Comment by Mike-DE-RE** [comment]
"IF YOUR ENTIRE APPLICATION IS LICENSED TO THE PUBLIC UNDER EITHER THE AFPL OR THE GNU AGPL, YOU CAN DISTRIBUTE AFPL GHOSTSCRIPT OR GNU GHOSTSCRIPT, RESPECTIVELY, WITH YOUR APPLICATION WITHOUT A COMMERCIAL LICENSE . . ."

https://ghostscript.com/doc/current/Commprod.htm

**Comment by RWAP** [comment]
Unfortunately, we do not plan on releasing the retro-printer source code and hardware designs under the GPL license.  I guess we could ask them, but I doubt they would see this as non-commercial.

Use for PrinterToPDF (the GPL version of the conversion code is different).

**Comment by Mike-DE-RE** [comment]
Are you sure you would have to release both (source code and hardware designs). As far as I know Linksys uses GPL Software as well for some of their routers. Are you sure that requires that they disclose hardware designs?
